Documentation/ceph-object-bucket-claim.md

```diff
@@ -0,0 +1,241 @@
+
+## Sample
+
+##### OBC Custom Resource (User Defined)
+```yaml
+apiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucketClaim
+metadata:
+  name: ceph-bucket [1]
+  namespace: rook-ceph [2]
+spec:
+  bucketName: [3]
+  generateBucketName: photo-booth [4]
+  storageClassName: rook-ceph-bucket [5]
+  additionalConfig: [6]
+    ANY_KEY: VALUE ...
+```
+1. `name` of the `ObjectBucketClaim`. This name becomes the name of the Secret and ConfigMap.
+1. `namespace`(optional) of the `ObjectBucketClaim`, which is also the namespace of the ConfigMap and Secret. If nothing is specified it uses `default`
+1. `name` of the `bucket`. If supplied then `generateBucketName` is ignored.
+**Not** recommended for new buckets since names must be unique within
+an entire object store.
+1. if supplied then `bucketName` must be empty. This value becomes the prefix for a randomly generated name.
+If both `bucketName` and `generateBucketName` are supplied then `BucketName` has precedence and `GenerateBucketName` is ignored.
+If both `bucketName` and `generateBucketName` are blank or omitted then the storage class is expected to contain the name of an _existing_ bucket. It's an error if all three bucket related names are blank or omitted.
+1. `storageClassName` which defines the StorageClass which contains the names of the bucket provisioner, the object-store and specifies the bucket retention policy.
+1. `additionalConfig` is an optional list of key-value pairs used to define attributes specific to the bucket being provisioned by this OBC. This information is typically tuned to a particular bucket provisioner and my limit application portability. Examples can include config values such as tenant, user and policy settings, etc.
+
+##### OBC Custom Resource  after Bucket Provisioning
+```yaml
+apiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucketClaim
+metadata:
+  finalizers: [1]
+  - objectbucket.io/finalizer
+  labels: [2]
+    bucket-provisioner: aws-s3.io-bucket [3]
+spec:
+  bucketName: photo-booth-62PrQ [4]
+  objectBucketRef: objectReference{} [5]
+  configMapRef: objectReference{} [6]
+  secretRef: objectReference{} [7]
+status:
+  phase: {"Pending", "Bound", "Released", "Failed"} [8]
+```
+1. the `finalizer` added by the provisioner protects the OBC from being deleted separately from the generated Secret and ConfigMap, the name is a constant.
+1. the `label` value shown is the name of the provisioner. In this example the provisioner name is aws-s3.io/bucket. This `label` is added to the `OBC`, `Secret`, `ConfigMap`, and `OB`. Note that the "/" in the provisioner name has been replaced by a "-" to conform to Kubernetes label values.
```

Comment:
>  to conform to Kubernetes label values.

It's a hard requirement that will cause failures, so it'd be good to word this more strongly.  e.g.

> to conform to Kubernetes label formatting requirements

Reply:
removed in the new version